### PR TITLE
Branch modify member

### LINKED ISSUE #68: Integrate save session framework with session command

### PR DESCRIPTION
**Pull request by keithxun**
Modify member and sessions class to be a one to many relationship.
Only add_session command is implemented.
However, note that UI is not updating as of now.
Storage is also not implemented yet.

### CODE REVIEW

File: src/main/java/hallpointer/address/logic/parser/EditSessionCommandParser.java
Comment (taggyhan) [inline src/main/java/hallpointer/address/logic/parser/EditSessionCommandParser.java:1]: shld be update>?

File: src/main/java/hallpointer/address/model/member/Member.java [hunk 1]
```diff
@@ -71,6 +106,46 @@ public Set<Tag> getTags() {
...
+    }
+
+    /**
+     *  Adds the given points to the member's total points.
+     *
+     *  @param points Points to be added to the member.
+     */
+    public void addPoints(Point points) {
+        requireNonNull(points);
+        this.totalPoints = new Point(this.totalPoints.points + points.points);
```
Comment (QinHaichen12) [inline src/main/java/hallpointer/address/model/member/Member.java]: can consider encapsulating this method in points. Tell don't ask. Also if this method is not used outside of addSession maybe can make it private?

File: src/main/java/hallpointer/address/model/member/Member.java [hunk 2]
```diff
@@ -71,6 +106,46 @@ public Set<Tag> getTags() {
...
+
+    /**
+     *  Adds the given session to the member's list of sessions.
+     *
+     *  @param session Session to be added to the member.
+     */
+    public void addSession(Session session) {
+        requireNonNull(session);
+        this.sessions.add(session);
+        this.totalPoints = new Point(this.totalPoints.points + session.getPoints().points);
```
Comment (QinHaichen12) [inline src/main/java/hallpointer/address/model/member/Member.java]: shouldn't this use addPoint method

File: src/main/java/hallpointer/address/model/member/Member.java [hunk 3]
```diff
@@ -71,6 +106,46 @@ public Set<Tag> getTags() {
...
+
+    /**
+     * Removes the given session from the member's list of sessions.
+     *
+     * @param session Session to be removed from the member.
+     */
+    public void removeSession(Session session) {
+        requireNonNull(session);
+        this.sessions.remove(session);
+        this.totalPoints = new Point(this.totalPoints.points - session.getPoints().points);
```
Comment (QinHaichen12) [inline src/main/java/hallpointer/address/model/member/Member.java]: implement minus point or just one method called changePoint?

File: src/main/java/hallpointer/address/model/point/Point.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    public int getValue() {
+        return points;
+    }
+
+    /**
+     * Returns the points as a formatted string.
+     */
+    @Override
+    public String toString() {
```
Comment (QinHaichen12) [inline src/main/java/hallpointer/address/model/point/Point.java:74]: more meaningful outputs like "3 Points"

File: src/main/java/hallpointer/address/model/member/Member.java (hunk 3 above)
Comment (QinHaichen12) [reply src/main/java/hallpointer/address/model/member/Member.java]: Also in what case will this method be called?

File: src/main/java/hallpointer/address/logic/commands/UpdateMemberCommand.java [hunk 5]
```diff
@@ -99,8 +101,10 @@ private static Member createUpdatedMember(Member memberToUpdate, UpdateMemberDes
         Telegram updatedTelegram = updateMemberDescriptor.getTelegram().orElse(memberToUpdate.getTelegram());
         Room updatedRoom = updateMemberDescriptor.getRoom().orElse(memberToUpdate.getRoom());
         Set<Tag> updatedTags = updateMemberDescriptor.getTags().orElse(memberToUpdate.getTags());
+        Point updatedPoints = updateMemberDescriptor.getTotalPoints().orElse(memberToUpdate.getTotalPoints());
```
Comment (QinHaichen12) [inline src/main/java/hallpointer/address/logic/commands/UpdateMemberCommand.java:104]: should we allow updatemember to interfere with points or sessions?

**Review by QinHaichen12 — COMMENTED** [review]
LGTM! Let me know if this is suitable to be merged and we should discuss class interactions later

File: src/main/java/hallpointer/address/logic/commands/AddSessionCommand.java [hunk 6]
```diff
@@ -46,9 +59,23 @@ public CommandResult execute(Model model) throws CommandException {
...
+        for (Index index : memberIndexes) {
+            if (index.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(MESSAGE_INVALID_INDEX);
+            }
+            Member member = lastShownList.get(index.getZeroBased());
+
+            member.addSession(toAdd);
+        }
+
         model.addSession(toAdd);
```
Comment (QinHaichen12) [inline src/main/java/hallpointer/address/logic/commands/AddSessionCommand.java]: What is the point of adding a session to model/addressbook if we are not going to display session? Should we remove all the session related methods in interface Model?

**Review by QinHaichen12 — COMMENTED** [review]
Is there a need to link model with sessions?

File: src/main/java/hallpointer/address/logic/commands/AddSessionCommand.java (hunk 6 above)
Comment (keithxun) [reply src/main/java/hallpointer/address/logic/commands/AddSessionCommand.java]: yea you are right, was just following the previous structure ngl

File: src/main/java/hallpointer/address/logic/parser/EditSessionCommandParser.java
Comment (keithxun) [reply src/main/java/hallpointer/address/logic/parser/EditSessionCommandParser.java:1]: uh this one for session command not update and following the other styles i think is member ah

File: src/main/java/hallpointer/address/model/member/Member.java (hunk 1 above)
Comment (keithxun) [reply src/main/java/hallpointer/address/model/member/Member.java]: i was thinking to make it public for the test case. Agree on the tell dont ask part

File: src/main/java/hallpointer/address/model/member/Member.java (hunk 3 above)
Comment (keithxun) [reply src/main/java/hallpointer/address/model/member/Member.java]: this should only be called if you delete the session. Will modify it to change point

File: src/main/java/hallpointer/address/model/point/Point.java (hunk 4 above)
Comment (keithxun) [reply src/main/java/hallpointer/address/model/point/Point.java:74]: agree

File: src/main/java/hallpointer/address/logic/commands/UpdateMemberCommand.java (hunk 5 above)
Comment (keithxun) [reply src/main/java/hallpointer/address/logic/commands/UpdateMemberCommand.java:104]: I dont think so? Command line for that will be difficult

File: src/main/java/hallpointer/address/model/member/Member.java (hunk 2 above)
Comment (keithxun) [reply src/main/java/hallpointer/address/model/member/Member.java]: agree

File: src/main/java/hallpointer/address/model/member/Member.java (hunk 3 above)
Comment (keithxun) [reply src/main/java/hallpointer/address/model/member/Member.java]: implemented subtractpoint. I dont think change point will be valid as points need to be positive

**Review by QinHaichen12 — COMMENTED** [review]
LGTM! Will modify model class after implementation is discussed later today.

**Pull request merged**